File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] freq = new int[26];
        int maxFreq = 0;
        int left = 0;
        int res = 0;
        // correct windows size = (right -left +1) - maxFreq = 1

        for (int right = 0; right < s.length(); right++) {
            char c = s.charAt(right);
            freq[c - 'A'] += 1;
            maxFreq = Math.max(maxFreq, freq[c - 'A']);

            while ((right - left + 1) - maxFreq > k) {
                freq[s.charAt(left) - 'A']--;
                left++;
            }
            res = Math.max(res, right - left + 1);
        }

        return res;
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        max = len(nums)/2

        dict1 = {}

        for i in nums:
            dict1[i] = dict1.get(i, 0) + 1
            if dict1[i] > max:
                return i

        return None